doc/sphinx/addendum/extraction.rst

```diff
@@ -399,6 +399,109 @@ As an example of translation to a non-inductive datatype, let's turn
 
    Extract Inductive nat => int [ "0" "succ" ] "(fun fO fS n -> if n=0 then fO () else fS (n-1))".
 
+Generating FFI Code
+~~~~~~~~~~~~~~~~~~~
+
+The plugin provides mechanisms to generate (currently only) OCaml code to
+interface the generated OCaml code with C programs. In order to link compiled
+OCaml code with C code, the linker needs to know
+
+ * which C functions will be called by the ML code (externals)
+ * which ML functions shall be accessible by the C code (callbacks)
+
+.. cmd:: Extract External Constant @qualid => @string
+
+   Same as the :cmd:`Extract Constant`, except that the given ML terms
+   will not be declared via ``let`` but in the form
+   ``external @qualid : ML type = "@string"``.
+   Also, no ``val`` declaration will be generated in the respective
+   mli file.
+
+.. caution::
+
+   * The external function name provided by the user as :n:`@string`
+     is currently **not** checked at all by extraction, even for syntax errors.
+
+   * It is the responsibility of the user to ensure that the C
+     functions given to realize the axioms do have the expected or compatible
+     types. In fact, the strings containing realizing code are just copied to the
+     extracted files.
+
+   * A previously used :cmd:`Extract Constant` for :n:`@qualid` will be overridden
+     by the command :cmd:`Extract External Constant`, although internally, the
+     specified :n:`@qualid` will still be part of the (inline) constant extraction
+     table. This is by design, since, when defining a function to be external, a
+     ``val`` declaration and ``let`` definition would collide with the external declaration
+     during compilation. Also, an external function cannot be inlined. Thus, the
+     command :cmd:`Extract Inlined Constant` has to be overridden, too.
+
+.. note::
+   Since the generated external expression contains the mapped OCaml type, the
+   C/C++ compiler is able to determine whether the ML function and respective
+   C/C++ function match in the arity and whether the types are compatible. Thus,
+   the type safety is not undermined when using :cmd:`Extract External Constant`.
```

Comment:
This whole paragraph is completely wrong; please remove it. Indeed, there is absolutely no check whatsoever, not even on the arity of the function. As long as there is a symbol corresponding to the string, the compilation/link will succeed it.

Reply:
Yes, it is. I tested that for passing not enough and too much parameters to an external function. The linker indeed doesn't care.